core/impl/src/main/java/com/blazebit/persistence/impl/ConstantifiedJoinNodeAttributeCollector.java

```diff
@@ -376,9 +376,17 @@ public void visit(EqPredicate predicate) {
             // TODO: at some point we should build an equivalence class to transitively propagate constantification
             // TODO: also, it would be nice if we could detect constantifications in disjuncts and work with that
             if (isConstant(predicate.getLeft())) {
-                predicate.getRight().accept(this);
+                if (isParameterOrLiteral(predicate.getRight())) {
+                    predicate.getLeft().accept(this);
+                } else {
+                    predicate.getRight().accept(this);
```

Comment:
Why is this dependent on the lhs not being a parameter or literal?

Reply:
The idea of the visitor is to visit all PathExpression that should be considered constantified i.e. they are in an equality relationshipt with something that is constant. The problem is that a PathExpression might already be constant due to e.g. a parent being constantified. In such a case, we'd miss out on initializing the constantification information for that Path which is essential. Before, we didn't have that, but only considered literals and parameters. That's why we do a special check here.